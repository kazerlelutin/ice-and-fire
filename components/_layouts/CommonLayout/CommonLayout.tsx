import Header from '../../Header/Header'
import Seo from '../../_UI/Seo/Seo'
import classes from './CommonLayout.module.css'

interface Props {
  readonly children: JSX.Element | Array<JSX.Element>
  readonly title?: string
}

export default function CommonLayout({ children, title }: Props) {
  return (
    <div className={classes.container}>
      <Seo title={title} />
      <header className={classes.header}><Header/></header>
      <main className={classes.main}>{children}</main>
    </div>
  )
}
