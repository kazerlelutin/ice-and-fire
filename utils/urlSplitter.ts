//for extract id 
export default function urlSplitter(url:string):string {
  const splitUrl = url.split('/')
  return splitUrl.at(-1)
}