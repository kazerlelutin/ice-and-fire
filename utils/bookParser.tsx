import { IapiBook } from "../types/apiBook.interface";
import { IBook } from "../types/book.interface";
import urlSplitter from "./urlSplitter";

export default function bookParser(book:IapiBook):IBook{

  return {
    id:urlSplitter(book.url),
    isbn: book.isbn,
    name: book.name,
    released: book.released,
    numberOfPages: book.numberOfPages,
    //merge all type of characters, just IDs
    characters: [...book.characters, ...book.povCharacters].reduce(
      (previousValue, currentValue) => {
        const 
        characterId = urlSplitter(currentValue),
        isExist = previousValue.find((id) => id === characterId)
        if (!isExist) previousValue.push(characterId)
        return previousValue
      },
      []
    ),
  }
}