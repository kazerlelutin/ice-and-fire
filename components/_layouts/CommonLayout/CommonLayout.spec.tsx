import { afterEach, describe, expect, it, vi } from 'vitest'
import { cleanup } from '@testing-library/react'
import { render, screen } from '../../../utils/test-utils'
import CommonLayout from './CommonLayout'

describe('<CommonLayout />', () => {
  afterEach(cleanup)

  it('Check snapshot', async () => {
    const { asFragment } = render(
      <CommonLayout>
        <p>my content</p>
      </CommonLayout>
    )

    expect(await screen.findByText('my content')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
