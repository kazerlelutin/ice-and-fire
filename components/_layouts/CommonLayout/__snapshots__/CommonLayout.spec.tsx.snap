// Vitest Snapshot v1

exports[`<CommonLayout /> > Check snapshot 1`] = `
<DocumentFragment>
  <div
    class="_container_f3f7d0"
  >
    <header
      class="_header_f3f7d0"
    >
      <div
        class="_container_46c4bf"
      >
        <div
          class="_container_5bb206"
        >
          Ice & Fire
        </div>
      </div>
    </header>
    <main
      class="_main_f3f7d0"
    >
      <p>
        my content
      </p>
    </main>
  </div>
</DocumentFragment>
`;
